### PR TITLE
run() interface become sync_run()/async_run(), add error handing, mod…

### PR DESCRIPTION
**Pull request by shangdehao1**
1: for server side,  Now, there have two interface : namely sync_run(),  async_run()

2: adding some error handing: when normally exit,  don't have any information output, for example EOF. otherwise output error information.

3: modify io_service_pool implement.

Simple testing have been done.

### CODE REVIEW

File: src/HDCSController.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ HDCSController::HDCSController(struct hdcs_repl_options repl_opt, std::string co
   network_service = new networking::server("0.0.0.0", _port, 16, 5);
   network_service->start([&](void* p, std::string s){handle_request(p, s);});
   //network_service->wait();
-  network_service->run();
+  network_service->sync_run();
 }
 
 HDCSController::~HDCSController() {
```
Comment (xuechendi) [inline src/HDCSController.cpp]: add network_service->stop() here?

Comment (shangdehao1) [reply src/HDCSController.cpp]: stop() interface have been added. 
when calling network_servive->stop(), two things will be done. 
1: close socket.
2: stop thread pool of network layer.

File: general.conf [hunk 2]
```diff
@@ -5,7 +5,7 @@ rbd_pool_name=rbd
 policy_mode=tier
 cache_mode=write_back
 cache_total_size=10737418240
-cache_dir=/mnt/cache/
+cache_dir=/mnt/hdcs/
```
Comment (zhouyuan) [inline general.conf]: can you change back to /mnt/cache?  this breaks several func tests

Comment (shangdehao1) [reply general.conf]: got it

**Comment by shangdehao1** [comment]
remove all binary files, and modify /mnt/hdcs/ into /mnt/cache/

**Pull request merged**